### PR TITLE
Roster validation and <ErrorBoundary> impl

### PR DESCRIPTION
**Pull request by DaveJackson-PE**
Summary of changes
------------------

This PR covers the requirements for the following Asana task:

> [Better input validation and error messages on roster load](https://app.asana.com/0/1188940305683068/1199145504016473)

List of changes
---------------

- Added EDIPI validation - 10 digit integer allowing leading zeros
- Added `value.trim().length` check in addition to `value.length` to `canSave()`
- Implemented `<ErrorBoundary>` component

What code can others reuse?
---------------------------

- `<ErrorBoundary>` 

Should this pull request be tested any more or less than usual?
---------------------------------------------------------------
Usual regression testing on the Roster page for validation and error handling should be sufficient.

What are general test steps or cases?
-------------------------------------

1. On Roster page, click 'UPLOAD CSV' button 
2. Select an invalid file
3. Verify the following error dialog:

Roster-Upload-CSV-Error

4. Dismiss the dialog and the list is active again, allowing the user to recover and continue.

### CODE REVIEW

**Review by larskendall — CHANGES_REQUESTED** [review]
I wasn't able to get the dialog to show when I tried triggering an error from the roster edit dialog. Did you confirm it works as expected?

I think the error boundary isn't meant for showing dialogs anyways, so it's probably worth aborting that particular approach and just stripping the Dialog stuff out of ErrorBoundary. The example shown in the React docs shows it being used to replace the normal component tree with a basic component to show an error, so that you show the error right on the element where the normal component would have rendered.

Following the React example, I think the error boundary might be as simple as this...

```typescript
import React, { ReactNode } from 'react';

export interface ErrorBoundaryProps {
  children?: ReactNode,
}

export interface ErrorBoundaryState {
  hasError: boolean
}

export class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
  constructor(props: ErrorBoundaryProps) {
    super(props);
    this.state = {
      hasError: false,
    };
  }

  static getDerivedStateFromError() {
    // Update state so the next render will show the fallback UI.
    return {
      hasError: true,
    };
  }

  render() {
    if (this.state.hasError) {
      // Render fallback UI.
      return <h1>Something went wrong.</h1>;
    }

    return this.props.children;
  }
}
```

File: src/components/error-boundary/error-boundary.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    });
+  }
+
+  render() {
+
+    return (
+      <>
+        {this.props.children}
+        {this.state.error && (
+        <Dialog
```
[the remark this answers is not in the document]
Comment (DaveJackson-PE) [reply src/components/error-boundary/error-boundary.tsx:52]: Ahhh... yeah I had the linter format the document and it didn't catch that. It didn't look quite right, but I wasn't sure what was off about it. Thanks for the info!

File: src/components/app.tsx [hunk 2]
```diff
@@ -139,8 +140,8 @@ export const App = () => {
   }
 
   return (
-    <>
+    <ErrorBoundary>
```
[the remark this answers is not in the document]
Comment (DaveJackson-PE) [reply src/components/app.tsx]: Resolved with latest commit

**Comment by DaveJackson-PE** [comment]
Running through `<ErrorBoundary>` tests again in production mode to track down the remaining issue

**Comment by DaveJackson-PE** [comment]
I finally found a decent example of a reasonable `<ErrorBoundary>` implementation and test case. I modified it slightly to show a `<Snackbar><Alert>...` instead of a `<Dialog>` so we can notify the user of the error yet still allow them to recover gracefully. I have a simple test case I'll provide in a separate repo for reference.

**Review by larskendall — APPROVED** [review]
This looks good! I'll go ahead and merge it.

**Pull request merged**